### PR TITLE
ghz: init at 0.109.0

### PR DESCRIPTION
**Pull request by zombiezen**
###### Description of changes

ghz is a gRPC benchmarking and load-testing tool: https://ghz.sh/

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/ghz/default.nix [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    owner = "bojand";
+    repo = "ghz";
+    rev = "v${version}";
+    sha256 = "sha256-5l2PeN+VxTaORAkmAfI9TCGd4W6y8BFs/eY4T9nYJuc=";
+  };
+  proxyVendor = true;
+  vendorSha256 = "sha256-R1OB4jTmyyS2O91vqSNjmSA19KutHzPLcLQ7R7tCsIA=";
+
+  subPackages = ["cmd/ghz" "cmd/ghz-web"];
+
```
Comment (azahi) [inline pkgs/tools/networking/ghz/default.nix:17]:
```suggestion

ldflags = [ "-s" "-w" ];

```

File: pkgs/tools/networking/ghz/default.nix [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  meta = {
```
Comment (azahi) [inline pkgs/tools/networking/ghz/default.nix:18]:
```suggestion
  meta = with lib; {
```
Please also remove other occurrences of `lib`.

File: pkgs/tools/networking/ghz/default.nix [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+buildGoModule rec {
+  pname = "ghz";
+  version = "0.109.0";
+
+  src = fetchFromGitHub {
+    owner = "bojand";
+    repo = "ghz";
+    rev = "v${version}";
+    sha256 = "sha256-5l2PeN+VxTaORAkmAfI9TCGd4W6y8BFs/eY4T9nYJuc=";
+  };
```
Comment (azahi) [inline pkgs/tools/networking/ghz/default.nix:12]:
```suggestion
  };
  
```

File: pkgs/tools/networking/ghz/default.nix [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  proxyVendor = true;
```
Comment (azahi) [inline pkgs/tools/networking/ghz/default.nix:13]: Are you sure `proxyVendor` is required here?
```suggestion
  proxyVendor = true;
  
```

**Review by azahi — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/ghz/default.nix (hunk 4 above)
Comment (zombiezen) [reply pkgs/tools/networking/ghz/default.nix:14]: I am not certain, but IIUC `proxyVendor` ensures that the checksum matches even when operating on case-insensitive filesystems. The [go.sum](https://github.com/bojand/ghz/blob/0d77cec5cfe83f13b3f5f877f9ab0a375369ecc9/go.sum) lists some dependencies with uppercase characters in the import path, so I figured I would proxy to be safe.

**Comment by zombiezen** [comment]
@azahi PTAL when you get the chance. Thank you!

**Review by azahi — CHANGES_REQUESTED** [review]
Seems like I forgot to press the "Submit review" button last time I've looked into this. My bad.

File: pkgs/tools/networking/ghz/default.nix (hunk 4 above)
Comment (zombiezen) [reply pkgs/tools/networking/ghz/default.nix]: Removed.

**Review by azahi — CHANGES_REQUESTED** [review]
Looks good. The only thing left now is to squash the commits. It also wouldn't hurt to rebase with the master branch.

**Comment by zombiezen** [comment]
Won't the commits get squashed as part of doing the merge? (I've contributed to nixpkgs a few times previously and never had to do this before. Just trying to figure out processes around here so I can reduce friction for future contributions.)

**Comment by azahi** [comment]
I'm not exactly sure but I guess it depends on which button committer presses when merging the PR. I and many others usually just squash or keep only meaningful commits.

**Comment by SuperSandro2000** [comment]
I just squash merged it but it is nicer to have them squashed already so that it is not forgotten.

**Comment by zombiezen** [comment]
Sounds good. Thanks for letting me know; I'll do that in the future!

Might also be good to add that the contributing docs (CONTRIBUTING.md and/or [the manual](https://nixos.org/manual/nixpkgs/stable/#chap-submitting-changes)). I'd be happy to write something up if you want.

**Pull request merged**